### PR TITLE
[Cosmos] Update max_integrated_cache_staleness_in_ms to allow 0

### PR DESCRIPTION
**Pull request by JericHunter**
Fixing a small issue with the integrated cache so that it can allow 0 as a value.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/azure/cosmos/_base.py [hunk 1]
```diff
@@ -671,5 +671,5 @@ def create_scope_from_url(url):
 
 def validate_cache_staleness_value(max_integrated_cache_staleness):
     int(max_integrated_cache_staleness)  # Will throw error if data type cant be converted to int
-    if max_integrated_cache_staleness <= 0:
+    if max_integrated_cache_staleness < 0:
         raise ValueError("Parameter 'max_integrated_cache_staleness_in_ms' can only be a positive integer.")
```
Comment (simorenoh) [inline sdk/cosmos/azure-cosmos/azure/cosmos/_base.py:675]: this message needs to be changed as well, since 0 is also accepted now

**Review by simorenoh — CHANGES_REQUESTED** [review]
please update any mentions of value needing to be positive in README (if any), update CHANGELOG with these changes under bugfixes section, and add a relevant tests for setting 0 for integrated cache

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@
 
 #### Bugs Fixed
 - Fixed parsing of args for overloaded `container.read()` method.
+- Fixed `validate_cache_staleness_value()` method to allow max_integrated_cache_staleness to be an integer greater than or equal to 0
```
Comment (simorenoh) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:12]: please add the PR link here, other than that looks great thanks Jeric!

**Review by simorenoh — APPROVED** [review]

**Review by simorenoh — CHANGES_REQUESTED** [review]
please fix the relevant tests for negative and 0 value to verify, the way the other tests do

File: sdk/cosmos/azure-cosmos/azure/cosmos/_base.py [hunk 3]
```diff
@@ -671,5 +671,6 @@ def create_scope_from_url(url):
 
 def validate_cache_staleness_value(max_integrated_cache_staleness):
     int(max_integrated_cache_staleness)  # Will throw error if data type cant be converted to int
-    if max_integrated_cache_staleness <= 0:
-        raise ValueError("Parameter 'max_integrated_cache_staleness_in_ms' can only be a positive integer.")
+    if max_integrated_cache_staleness < 0:
+        raise ValueError("Parameter 'max_integrated_cache_staleness_in_ms' can only be an "
+                         "integer greater than or equal to zero")
```
Comment (tjprescott) [inline sdk/cosmos/azure-cosmos/azure/cosmos/_base.py:676]: Why are we doing client-side validation here instead of letting the service throw an error?

Comment (simorenoh) [reply sdk/cosmos/azure-cosmos/azure/cosmos/_base.py:676]: I believe a negative value still gets accepted by the service but gets ignored altogether - this way the user knows the ranges of numbers they can provide
@JericHunter can you please remove this validation and double check that this is the case?

Comment (tjprescott) [reply sdk/cosmos/azure-cosmos/azure/cosmos/_base.py:676]: If the service provides a good error, we should just remove this CS validation. If indeed it does simply ignore it, then this is good.

Comment (JericHunter) [reply sdk/cosmos/azure-cosmos/azure/cosmos/_base.py:676]: The service does not provide a good error so we should keep this

**Review by simorenoh — APPROVED** [review]

**Pull request merged**